docs/proposals/volumes.md

```diff
@@ -0,0 +1,510 @@
+not yet been accepted.
+
+### Rocket
+
+Rocket
+[image manifests](https://github.com/appc/spec/blob/master/spec/aci.md#image-manifest-schema) can
+specify users and groups, similarly to how a Docker image can.  A Rocket
+[pod manifest](https://github.com/appc/spec/blob/master/spec/pods.md#pod-manifest-schema) can also
+override the default user and group specified by the image manifest.
+
+Rocket does not currently support supplemental groups or changing the owning UID or
+group of a volume, but it has been [requested](https://github.com/coreos/rkt/issues/1309).
+
+## Use Cases
+
+1.  As a user, I want the system to set ownership and permissions on volumes correctly to enable
+    reads and writes with the following scenarios:
+    1.  All containers running as root
+    2.  All containers running as the same non-root user
+    3.  Multiple containers running as a mix of root and non-root users
+
+### All containers running as root
+
+For volumes that only need to be used by root, no action needs to be taken to change ownership or
+permissions, but setting the ownership based on the supplemental group shared by all containers in a
+pod will also work.  For situations where read-only access to a shared volume is required from one
+or more containers, the `VolumeMount`s in those containers should have the `readOnly` field set.
+
+### All containers running as a single non-root user
+
+In use cases whether a volume is used by a single non-root UID the volume ownership and permissions
+should be set to enable read/write access.
+
+Currently, a non-root UID will not have permissions to write to any but an `EmptyDir` volume.
+Today, users that need this case to work can:
+
+1.  Grant the container the necessary capabilities to `chown` and `chmod` the volume:
+    - `CAP_FOWNER`
+    - `CAP_CHOWN`
+    - `CAP_DAC_OVERRIDE`
+2.  Run a wrapper script that runs `chown` and `chmod` commands to set the desired ownership and
+    permissions on the volume before starting their main process
+
+This workaround has significant drawbacks:
+
+1.  It grants powerful kernel capabilities to the code in the image and thus is not securing,
+    defeating the reason containers are run as non-root users
+2.  The user experience is poor; it requires changing Dockerfile, adding a layer, or modifying the
+    container's command
+
+Some cluster operators manage the ownership of shared storage volumes on the server side.
+In this scenario, the UID of the container using the volume is known in advance.  The ownership of
+the volume is set to match the container's UID on the server side.
+
+### Containers running as a mix of root and non-root users
+
+If the list of UIDs that need to use a volume includes both root and non-root users, supplemental
+groups can be applied to enable sharing volumes between containers.  The ownership and permissions
+`root:<supplemental group> 0770` will make a volume usable from both containers running as root and
```

Comment:
I'm going to say we want the setgid bit sit for now.  I'm still thinking through all the implications of sticky bit.  I think we want it, trying to determine if we should set it on all volumes.